spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/DecisionTreeTrainer.java

```diff
@@ -0,0 +1,259 @@
+import co.cask.cdap.etl.api.batch.SparkPluginContext;
+import co.cask.cdap.etl.api.batch.SparkSink;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Splitter;
+import com.google.common.primitives.Doubles;
+import joptsimple.internal.Strings;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.api.java.function.Function;
+import org.apache.spark.mllib.linalg.Vectors;
+import org.apache.spark.mllib.regression.LabeledPoint;
+import org.apache.spark.mllib.tree.DecisionTree;
+import org.apache.spark.mllib.tree.model.DecisionTreeModel;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Spark Sink plugin that trains a model based upon a label in the structured record using Decision Tree Regression.
+ * Writes this model to a FileSet.
+ */
+@Plugin(type = SparkSink.PLUGIN_TYPE)
+@Name(DecisionTreeTrainer.PLUGIN_NAME)
+@Description("Trains a regression model based upon a particular label and featuresToInclude of a record.")
+public class DecisionTreeTrainer extends SparkSink<StructuredRecord> {
+  public static final String PLUGIN_NAME = "DecisionTreeTrainer";
+  //Impurity measure of the homogeneity of the labels at the node. Expected value for regression is "variance".
+  private static final String IMPURITY = "variance";
+  private static final int DEFAULT_MAX_DEPTH = 10;
+  private static final int DEFAULT_MAX_BINS = 100;
+  private DecisionTreeTrainerConfig config;
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    pipelineConfigurer.createDataset(config.fileSetName, FileSet.class);
+    Schema inputSchema = pipelineConfigurer.getStageConfigurer().getInputSchema();
+    if (inputSchema != null) {
+      config.validate(inputSchema);
+    }
+  }
+
+  @Override
+  public void run(SparkExecutionPluginContext context, final JavaRDD<StructuredRecord> input) throws Exception {
+
+    Schema schema = input.first().getSchema();
+    //categoricalFeaturesInfo : Specifies which features are categorical and how many categorical values each of those
+    //features can take. This is given as a map from feature index to the number of categories for that feature.
+    //Empty categoricalFeaturesInfo indicates all features are continuous.
+    Map<Integer, Integer> categoricalFeaturesInfo = config.getCategoricalFeatureInfo(schema);
+    final List<String> fields = config.getFeatureList(schema);
+
+    JavaRDD<LabeledPoint> trainingData = input.map(new Function<StructuredRecord, LabeledPoint>() {
+      @Override
+      public LabeledPoint call(StructuredRecord record) throws Exception {
+        List<Double> featureList = new ArrayList<>();
+        for (String field : fields) {
+          featureList.add(Double.valueOf(record.get(field).toString()));
```

Comment:
need to handle the case when record.get(field) is null. This will be common, as many vectors are usually sparse.